hls-graph/src/Development/IDE/Graph/Internal/Database.hs

```diff
@@ -93,7 +94,7 @@ builder db@Database{..} keys = withRunInIO $ \(RunInIO run) -> do
     -- Things that I need to force before my results are ready
     toForce <- liftIO $ newTVarIO []
     current <- liftIO $ readTVarIO databaseStep
-    results <- liftIO $ atomically $ for keys $ \id -> do
+    results <- liftIO $ for keys $ \id -> atomicallyNamed "builder" $ do
```

Comment:
I was just thinking a few notes like this.
```suggestion
    -- Run the STM transaction *inside* the loop so we get smaller transactions and less contention
    results <- liftIO $ for keys $ \id -> atomicallyNamed "builder" $ do
```

Reply:
Added, thanks!